test/util/helper.go

```diff
@@ -148,5 +148,5 @@ func TestCertificate(dnsName string, resp *http.Response) error {
 			}
 		}
 	}
-	return fmt.Errorf("wildcard hostname not found in the certificate via get request")
+	return fmt.Errorf("%s not found in the certificate via get request", dnsName)
```

Comment:
Should we just remove all usages of this TestCertificate function i.e. https://github.com/Kuadrant/multicluster-gateway-controller/blob/3dfbf32c018cbfcddd8d036763c8dbf71ef01a08/test/e2e/gateway_single_spoke_test.go#L354